minijinja/src/value/mod.rs

```diff
@@ -930,15 +1000,26 @@ impl Serialize for Value {
                 }
                 map.end()
             }
-            ValueRepr::Dynamic(ref n) => {
-                use serde::ser::SerializeMap;
-                let mut s = ok!(serializer.serialize_map(None));
-                for k in n.attributes() {
-                    let v = n.get_attr(k).unwrap_or(Value::UNDEFINED);
-                    ok!(s.serialize_entry(&k, &v));
+            ValueRepr::Dynamic(ref dy) => match dy.behavior() {
+                ObjectBehavior::Basic => serializer.serialize_str(&dy.to_string()),
+                ObjectBehavior::Seq(s) => {
+                    use serde::ser::SerializeSeq;
+                    let mut seq = ok!(serializer.serialize_seq(Some(s.len())));
+                    for idx in 0..s.len() {
+                        ok!(seq.serialize_element(&s.get(idx).unwrap_or(Value::UNDEFINED)));
```

Comment:
Should this bail on the first `None`? I suppose it's odd for the `AsSeq` to return a `len()` where every `0..len()` isn't included, but still. It would be nice to clarify the spec for `AsSeq`.

Reply:
I agree that this should be clarified in the trait. I think the most sensible definition is that you should not lie about your `len` but that if you do, the engine should assume `UNDEFINED`. This is also the semantics that arrays in JavaScript have.